moto/sagemaker/models.py

```diff
@@ -1890,6 +1928,38 @@ def list_training_jobs(
             "NextToken": str(next_index) if next_index is not None else None,
         }
 
+    def update_endpoint_weights_and_capacities(
+        self, endpoint_name, desired_weights_and_capacities
+    ):
+
+        endpoint = self.endpoints.get(endpoint_name, None)
+        if not endpoint:
+            raise AWSValidationException(
+                f'Could not find endpoint "{FakeEndpoint.arn_formatter(endpoint_name, self.region_name)}".'
+            )
+
+        endpoint.endpoint_status = "Updating"
+
+        for variant_config in desired_weights_and_capacities:
+            name = variant_config.get("VariantName")
+            desired_weight = variant_config.get("DesiredWeight")
+            desired_instance_count = variant_config.get("DesiredInstanceCount")
+
+            for index, variant in enumerate(endpoint.production_variants):
+                if variant.get("VariantName") == name:
+                    variant["DesiredWeight"] = desired_weight
+                    variant["CurrentWeight"] = desired_weight
+                    variant["DesiredInstanceCount"] = desired_instance_count
+                    variant["CurrentInstanceCount"] = desired_instance_count
+                    break
+                elif index == len(endpoint.production_variants) - 1:
+                    raise AWSValidationException(
```

Comment:
If there are two variants, and the user wants to update 3 variants where the second one doesn't exist, this would update the first, but throw an error on the second - and completely ignore the third. Am I understanding that correctly? Is that how AWS behaves?

Normally a check like that happens in the beginning, but it wouldn't surprise me if this is how AWS implements it..